File: Pythoncode/55.跳跃游戏.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        '''
        https://www.cnblogs.com/grandyang/p/4371526.html
        '''
        reach = 0 + nums[0]
        for i in range(len(nums)):
            if i > reach:
                return False
            else:
                reach = max(reach, i + nums[i])
        return True

File: Cppcode/55.跳跃游戏.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxReachPoint = 0;
        int i = 0;
        while (i <= maxReachPoint)
        {
            if (i == nums.size()-1) return true;
            maxReachPoint = max(maxReachPoint, i+nums[i]);
            i++;
        }
        return false;
    }
};